### PR TITLE
[Merged by Bors] - TY-2957 post endpoint

### PR DESCRIPTION
**Pull request by rustonaut**
Allow using POST for semantic GET requests on some endpoints to work around limitations of some software in the pipeline which limits the length of the GET query.

The query parameters will be added to as a json body mapping query parameter names to values.

This currently adds all values as Strings even numbers, this currently works, but we might want to consider fixing it, but doing so adds quite a bunch of code.

**References:**

- [TY-2957](https://xainag.atlassian.net/browse/TY-2957)
- based on #462

### CODE REVIEW

**Comment by rustonaut** [comment]
One question is if we should make it configurable or hard code weather `get_as_post` is used. 

Currently I just add the option to use it, but do not trigger the option. I have a local ad-hoc branch for testing which also contains e.g. unmerged changes from #462 and some ad-hoc script I will dump somewhere (news garden?).

**Comment by acrrd** [comment]
> One question is if we should make it configurable or hard code weather `get_as_post` is used.
> 

After a bit of testing, if everything is fine we are going to always use post.

**Review by finiteprods — APPROVED** [review]
looks good, included some trivial comments.

File: discovery_engine_core/providers/src/helpers/rest_endpoint.rs [hunk 1]
```diff
@@ -60,25 +63,50 @@ impl RestEndpoint {
         self
     }
 
+    /// Configures if we should use POST for GET requests.
+    ///
+    /// This is sometimes needed when some part of the serve
+    /// pipeline puts to strict limits on the length of the
+    /// path/query.
```
[the remark this answers is not in the document]
Comment (rustonaut) [reply discovery_engine_core/providers/src/helpers/rest_endpoint.rs:70]: server pipeline, i.e. some software they use for processing the get requests is trimming them.

That could be an API gateway, nginx, their endpoints implementation etc.

I changed it to just say "server" dropping the pipeline part. From our POV they are all the same opaque entitiy anyway.

**Comment by rustonaut** [comment]
I also like discussed added TY-3006, i.e. I enabled it for all newscatcher endpoints.

**Comment by rustonaut** [comment]
oh, right it's failing because the tests expect get :facepalm:

**Comment by rustonaut** [comment]
ok, removed TY-3006 parts as mentioned in the daily

**Comment by rustonaut** [comment]
@finiteprods can I go ahead an merge it. It now should only contain the review changes.

**Review by finiteprods — APPROVED** [review]

**Comment by rustonaut** [comment]
bors merge

**Pull request closed without merging**